lib/algo/DMD.cpp

```diff
@@ -125,6 +125,54 @@ DMD::DMD(std::vector<std::complex<double>> eigs, Matrix* phi_real,
     setOffset(state_offset, 0);
 }
 
+DMD::~DMD()
+{
+    for (auto snapshot : d_snapshots)
+    {
+        delete snapshot;
+    }
+    for (auto sampled_time : d_sampled_times)
+    {
+        delete sampled_time;
+    }
+    if (d_state_offset != nullptr)
+    {
+        delete d_state_offset;
+    }
+    if (d_basis != nullptr)
+    {
+        delete d_basis;
+    }
+    if (d_A_tilde != nullptr)
+    {
+        delete d_A_tilde;
+    }
+    if (d_phi_real != nullptr)
```

Comment:
Nit: A nullptr in C++ is always falsy in boolean contexts, so e.g. if(d_phi_real) {...} should suffice here.

Reply:
Deleting `nullptr` is safe, but based on some internet discussions I'm not sure whether it is safe in more recent versions of C++. So it may be ok to remove the checks for `nullptr` and just delete. Otherwise, the suggestion by @pranav-vempati sounds like a good change to make here.

Reply:
Also, note that DMD.h initializes to `NULL` rather than `nullptr`.

Reply:
Thanks for the comments, guys.  I couldn't find any discussion on whether `delete`ing a `nullptr` is not safe, so I opted to get rid of the `if` statements.  Would targeting a specific version of C++ in `compile.sh` (say C++11 or 14?) help avoid this problem @dylan-copeland?